### PR TITLE
Refactor test-balance service

### PR DESCRIPTION
**Pull request by osipov-mit**

### CODE REVIEW

File: idea/common/src/jsonrpc-errors.ts [hunk 1]
```diff
@@ -99,6 +99,16 @@ export const JSONRPC_ERRORS = {
...
     message: 'Forbidden',
   },
+  InvalidAddress: {
+    name: 'InvalidAddress',
+    code: -32400,
+    message: 'Address is invalid',
+  },
+  NotSupportedAddress: {
+    name: 'NotSupportedAddress',
+    code: -32400,
```
Comment (ElessarST) [inline idea/common/src/jsonrpc-errors.ts:109]: maybe it's better have different codes for different errors?

File: idea/test-balance/src/services/gear.ts [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    await new Promise((resolve) => {
+      batch
+        .signAndSend(this.account, ({ events, status }) => {
+          if (status.isInBlock) {
+            blockHash = status.asInBlock.toHex();
+            logger.info(`Batch is in block`, { blockHash, correlationId });
+
+            for (const { event } of events) {
+              switch (event.method) {
+                case 'Transfer':
```
Comment (ElessarST) [inline idea/test-balance/src/services/gear.ts:106]: Transfer, ExtrinsicSuccess ... can be moved to enum

File: idea/test-balance/src/services/gear.ts [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    let blockHash: string;
+
+    const correlationId = randomUUID({});
+
+    logger.info(`Sending batch with ${addresses.length} transfers`, { addresses, correlationId });
+
+    await new Promise((resolve) => {
+      batch
+        .signAndSend(this.account, ({ events, status }) => {
+          if (status.isInBlock) {
```
Comment (ElessarST) [inline idea/test-balance/src/services/gear.ts:100]: nit: early return can be used here
```
if (!status.isInBlock) {
  return;
}
...
```

File: idea/test-balance/src/services/gear.ts [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+            logger.info(`Batch is in block`, { blockHash, correlationId });
+
+            for (const { event } of events) {
+              switch (event.method) {
+                case 'Transfer':
+                  transferred.push((event.data as TransferData).to.toHex());
+                  break;
+                case 'ExtrinsicSuccess':
+                  logger.info(`ExtrinsicSuccess`, { blockHash, correlationId });
+                  resolve(0);
```
Comment (ElessarST) [inline idea/test-balance/src/services/gear.ts:111]: `0/1` can be moved to const with status names to better understanding

File: idea/common/src/jsonrpc-errors.ts (hunk 1 above)
Comment (osipov-mit) [reply idea/common/src/jsonrpc-errors.ts:109]: Yes, gonna do it in a different pr

File: idea/test-balance/src/services/gear.ts (hunk 2 above)
Comment (osipov-mit) [reply idea/test-balance/src/services/gear.ts]: refactored

File: idea/test-balance/src/services/gear.ts (hunk 3 above)
Comment (osipov-mit) [reply idea/test-balance/src/services/gear.ts]: thanks

File: idea/test-balance/src/services/gear.ts (hunk 4 above)
Comment (osipov-mit) [reply idea/test-balance/src/services/gear.ts]: refactored this part

**Review by ElessarST — APPROVED** [review]

**Pull request merged**